Demo/Application/Features/ShopperInsightsViewController.swift

```diff
@@ -0,0 +1,164 @@
+            [
+                shopperInsightsInputView.leadingAnchor.constraint(equalTo: view.leadingAnchor),
+                shopperInsightsInputView.trailingAnchor.constraint(equalTo: view.trailingAnchor),
+                shopperInsightsInputView.topAnchor.constraint(equalTo: view.topAnchor),
+                shopperInsightsInputView.widthAnchor.constraint(equalTo: view.widthAnchor),
+                shopperInsightsButton.leadingAnchor.constraint(equalTo: view.leadingAnchor, constant: 40),
+                shopperInsightsButton.topAnchor.constraint(equalTo: shopperInsightsInputView.bottomAnchor, constant: 10),
+                shopperInsightsButton.widthAnchor.constraint(
+                    equalTo: shopperInsightsInputView.widthAnchor,
+                    multiplier: 0.8
+                ),
+            ]
+        )
+    }
+    
+    override func createPaymentButton() -> UIView {
+        let buttons = [payPalVaultButton, venmoButton]
+        buttons.forEach { $0.isEnabled = false }
+        let stackView = UIStackView(arrangedSubviews: buttons)
+        stackView.axis = .vertical
+        stackView.alignment = .center
+        stackView.distribution = .fillEqually
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        
+        shopperInsightsClient.sendPayPalPresentedEvent()
+        shopperInsightsClient.sendVenmoPresentedEvent()
+        
+        return stackView
+    }
+    
+    @objc func shopperInsightsButtonTapped(_ button: UIButton) {
+        self.progressBlock("Fetching shopper insights...")
+        
+        let request = BTShopperInsightsRequest(
+            email: emailView.textField.text ?? "",
+            phone: Phone(
+                countryCode: countryCodeView.textField.text ?? "",
+                nationalNumber: nationalNumberView.textField.text ?? ""
+            )
+        )
+        Task {
+            do {
+                let result = try await shopperInsightsClient.getRecommendedPaymentMethods(request: request)
+                self.progressBlock("PayPal Recommended: \(result.isPayPalRecommended)\nVenmo Recommended: \(result.isVenmoRecommended)\nEligible in PayPal Network: \(result.isEligibleInPayPalNetwork)")
+                self.payPalVaultButton.isEnabled = result.isPayPalRecommended
+                self.venmoButton.isEnabled = result.isVenmoRecommended
+            } catch {
+                self.progressBlock("Error: \(error.localizedDescription)")
+            }
+        }
+    }
+    
+    @objc func payPalVaultButtonTapped(_ button: UIButton) {
+        self.progressBlock("Tapped PayPal Vault")
+        shopperInsightsClient.sendPayPalSelectedEvent()
+        
+        button.setTitle("Processing...", for: .disabled)
+        button.isEnabled = false
+        
+        let paypalRequest = BTPayPalVaultRequest()
+        paypalRequest.userAuthenticationEmail = emailView.textField.text ?? nil
+        
+        paypalClient.tokenize(paypalRequest) { (nonce, error) in
```

Comment:
```suggestion
        paypalClient.tokenize(paypalRequest) { nonce, error in
```

Reply:
Speaking of - I added a tickets for future us to move the Demo app & Unit Tests to use the SDKs async-await methods vs completion ones